tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -821,35 +848,505 @@ private void throwExceptionIfTypeIsIncompatible(Object o) {
     }
   }
 
-  private static native long allocate(int dtype, long[] shape, long byteSize);
+  private static void requireHandle(TF_Tensor handle) {
+    if (handle == null || handle.isNull()) {
+      throw new NullPointerException("close() was called on the Tensor");
+    }
+  }
+
+  private static int elemByteSize(int dtype) {
+    // The code in this file makes the assumption that the
+    // TensorFlow TF_DataTypes and the Java primitive types
+    // have the same byte sizes. Validate that:
+    switch (dtype) {
+      case TF_BOOL:
+      case TF_UINT8:
+        assert Loader.sizeof(BooleanPointer.class) == 1 :
+               "Java boolean not compatible with TF_BOOL";
+        assert Loader.sizeof(BytePointer.class) == 1 :
+               "Java byte not compatible with TF_UINT8";
+        return 1;
+      case TF_FLOAT:
+      case TF_INT32:
+        assert Loader.sizeof(FloatPointer.class) == 4 :
+               "Java float not compatible with TF_FLOAT";
+        assert Loader.sizeof(IntPointer.class) == 4 :
+               "Java int not compatible with TF_INT32";
+        return 4;
+      case TF_DOUBLE:
+      case TF_INT64:
+        assert Loader.sizeof(DoublePointer.class) == 8 :
+               "Java double not compatible with TF_DOUBLE";
+        assert Loader.sizeof(LongPointer.class) == 8 :
+               "Java long not compatible with TF_INT64";
+        return 8;
+      default:
+        return 0;
+    }
+  }
+
+  /** Write a Java scalar object (java.lang.Integer etc.) to a TF_Tensor. */
+  private static void writeScalar(Object src, int dtype, BytePointer dst, long dstSize) {
```

Comment:
Most of the stuff here and below in this class should ultimately be dropped and replaced with the usage of TF Tools to read or write data from/to the tensor.

I guess we can migrate it to JavaCPP meanwhile and I can take care of the cleanup after.